src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -0,0 +1,2164 @@
+                SetWinHttpOption(
+                    requestHandle,
+                    Interop.WinHttp.WINHTTP_OPTION_MAX_HTTP_AUTOMATIC_REDIRECTS,
+                    ref optionData);
+            }
+
+            optionData = (uint)_automaticRedirectionPolicy;
+            SetWinHttpOption(requestHandle, Interop.WinHttp.WINHTTP_OPTION_REDIRECT_POLICY, ref optionData);
+        }
+
+        private void SetRequestHandleCookieOptions(SafeInternetHandle requestHandle)
+        {
+            if (_cookieUsePolicy == CookieUsePolicy.UseSpecifiedCookieContainer ||
+                _cookieUsePolicy == CookieUsePolicy.IgnoreCookies)
+            {
+                uint optionData = Interop.WinHttp.WINHTTP_DISABLE_COOKIES;
+                SetWinHttpOption(requestHandle, Interop.WinHttp.WINHTTP_OPTION_DISABLE_FEATURE, ref optionData);
+            }
+        }
+
+        private void SetRequestHandleTlsOptions(SafeInternetHandle requestHandle)
+        {
+            // If we have a custom server certificate validation callback method then 
+            // we need to have WinHTTP ignore some errors so that the callback method
+            // will have a chance to be called.
+            uint optionData;
+            if (_serverCertificateValidationCallback != null)
+            {
+                optionData =
+                    Interop.WinHttp.SECURITY_FLAG_IGNORE_UNKNOWN_CA |
+                    Interop.WinHttp.SECURITY_FLAG_IGNORE_CERT_WRONG_USAGE |
+                    Interop.WinHttp.SECURITY_FLAG_IGNORE_CERT_CN_INVALID |
+                    Interop.WinHttp.SECURITY_FLAG_IGNORE_CERT_DATE_INVALID;
+                SetWinHttpOption(requestHandle, Interop.WinHttp.WINHTTP_OPTION_SECURITY_FLAGS, ref optionData);
+            }
+            else if (_checkCertificateRevocationList)
+            {
+                // If no custom validation method, then we let WinHTTP do the revocation check itself.
+                optionData = Interop.WinHttp.WINHTTP_ENABLE_SSL_REVOCATION;
+                SetWinHttpOption(requestHandle, Interop.WinHttp.WINHTTP_OPTION_ENABLE_FEATURE, ref optionData);
+            }
+        }
+
+        private void SetRequestHandleClientCertificateOptions(SafeInternetHandle requestHandle, Uri requestUri)
+        {
+            // Must be HTTPS scheme to use client certificates.
+            if (requestUri.Scheme != UriSchemeHttps)
+            {
+                return;
+            }
+
+            // Get candidate list for client certificates.
+            X509Certificate2Collection certs;
+            if (_clientCertificateOption == ClientCertificateOption.Manual)
+            {
+                certs = ClientCertificates;
+            }
+            else
+            {
+                var myStore = new X509Store();
+                myStore.Open(OpenFlags.OpenExistingOnly | OpenFlags.ReadOnly);
+                certs = myStore.Certificates;
+                myStore.Dispose();
```

Comment:
Do we need to be concerned about exceptions such that this should be:
```C#
using (var myStore = new X509Store())
{
    ...
}
```
instead of:
```C#
var myStore = new X509Store();
...
myStore.Dispose();
```
?